upload/upload.go

```
@@ -29,40 +47,48 @@ func CreateImageUploader(driver Driver, imgCfg ImageConfig) (*ImageUploader, err
 
 // Save saves the uploaded image, including generating a thumbnail,
 // to the upload store.
-func (u ImageUploader) Save(recipeID int64, imageName string, data []byte) (originalURL, thumbnailURL string, err error) {
+func (u ImageUploader) Save(recipeID int64, imageName string, data []byte) (result *SaveResult, err error) {
 	ok, contentType := isImageFile(data)
 	if !ok {
-		return "", "", fmt.Errorf("attachment must be an image; content type: %s ", contentType)
+		return nil, fmt.Errorf("attachment must be an image; content type: %s ", contentType)
 	}
 
+	// Make sure the file extension is .jpeg
+	imageExt := filepath.Ext(imageName)
+	imageName = imageName[0:len(imageName)-len(imageExt)] + ".jpeg"
+
 	// First decode the image
 	dataReader := bytes.NewReader(data)
-	original, err := imaging.Decode(dataReader, imaging.AutoOrientation(true))
+	original, format, err := image.Decode(dataReader)
+	// QUESTION: Do we need to auto-detect EXIF orientation and rotate the image accordingly?
 	if err != nil {
-		return "", "", fmt.Errorf("failed to decode image: %w", err)
+		return nil, fmt.Errorf("failed to decode image: %w", err)
 	}
 
-	// Then determine if it should be resized before saving
-	var origURL string
+	var imageURL string
 	imgDir := getDirPathForImage(recipeID)
-	if u.imgCfg.ImageQuality == ImageQualityOriginal {
+	if format == "jpeg" && u.imgCfg.ImageQuality == ImageQualityOriginal {
 		// Save the original as-is
-		origURL, err = u.saveImage(data, imgDir, imageName)
+		imageURL, err = u.saveImage(data, imgDir, imageName)
 	} else {
-		// Resize and save
-		origURL, err = u.generateFitted(original, contentType, imgDir, imageName)
-	}
-	if err != nil {
```

The error is being ignored for the original quality path due to this change